src/api/app/lib/backend/proxy_request.rb

```diff
@@ -0,0 +1,136 @@
+module Backend
+  # The backend shares (a subset of) our API routes. This is a service to proxy requests to those routes.
+  # It caches incoming/outgoing files in our tmp directory or makes use of the various forward/rewrite/redirect
+  # plugins for apache, nginx or lighttpd if configured.
+  class ProxyRequest
+    attr_accessor :path, :request
+
+    def initialize(request, path_override = nil)
+      @request = request
+      @path = path_override || path_from_request
+    end
+
+    def call
+      case request.method_symbol
+      when :get, :head
+        proxy_from_backend unless use_web_server_forward
+      when :post, :put, :delete
+        proxy_to_backend
+      end
+    end
+
+    private
+
+    def path_from_request
+      path = @request.path_info
+      query_string = @request.query_string
+      if @request.form_data?
+        # it's uncommon, but possible that we have both
+        query_string += '&' if query_string.present?
+        query_string += @request.raw_post
+      end
+      query_string = '?' + query_string if query_string.present?
+      path + query_string
+    end
+
+    def use_web_server_forward
+      # apache & mod_xforward case
+      if CONFIG['use_xforward'] && CONFIG['use_xforward'] != 'false'
+        logger.debug "[backend] VOLLEY(mod_xforward): #{@path}"
+        headers['X-Forward'] = "http://#{CONFIG['source_host']}:#{CONFIG['source_port']}#{@path}"
+        headers['Cache-Control'] = 'no-transform' # avoid compression
+        head(200)
+        @skip_validation = true
+        return true
+      end
+
+      # lighttpd 1.5 case
+      if CONFIG['x_rewrite_host']
+        logger.debug "[backend] VOLLEY(lighttpd): #{@path}"
+        headers['X-Rewrite-URI'] = @path
+        headers['X-Rewrite-Host'] = CONFIG['x_rewrite_host']
+        headers['Cache-Control'] = 'no-transform' # avoid compression
+        head(200)
+        @skip_validation = true
+        return true
+      end
+
+      # nginx case
+      if CONFIG['use_nginx_redirect']
+        logger.debug "[backend] VOLLEY(nginx): #{@path}"
+        headers['X-Accel-Redirect'] = "#{CONFIG['use_nginx_redirect']}/http/#{CONFIG['source_host']}:#{CONFIG['source_port']}#{@path}"
+        headers['Cache-Control'] = 'no-transform' # avoid compression
+        head(200)
+        @skip_validation = true
+        return true
+      end
+
+      false
+    end
+
+    # FIXME: This shares a lot of logic with Backend::File, should be merged.
+    def proxy_from_backend
+      logger.debug "[backend] VOLLEY: #{@path}"
+      Backend::Test.start
```

Comment:
`Backend::Test.start` as well in production?

Reply:
this method is doing too many things.. couldn't we break it in smaller methods?

Reply:
I try to limit the refactoring to clarification here.

Reply:
> Backend::Test.start as well in production?

First line i `Backend::Test.start` is `return unless Rails.env.test?`

Reply:
I would prefer to have the `if` inside of the method here to make it more explicit?

Reply:
I really try to limit the refactoring to clarification here. Otherwise this will be an endless rabbit hole. As you can see from the spec, this already leaked far into the app.